services/src/handlers/datasets.rs

```diff
@@ -144,13 +183,69 @@ async fn get_dataset_handler<C: Context>(
     Ok(web::Json(dataset))
 }
 
+/// Creates a new dataset using files available on the volumes.
+/// Requires an admin session.
+#[utoipa::path(
+    tag = "Datasets",
+    post,
+    // TODO: "public" doesn't really make sense, because there are no private datasets. But this is more consistent with pro version
+    path = "/dataset/public", 
+    request_body = CreateSystemDataset,
+    responses(
+        (status = 200, description = "OK", body = IdResponse,
+            example = json!({
+                "id": {
+                    "internal": "8d3471ab-fcf7-4c1b-bbc1-00477adf07c8"
+                }
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn create_system_dataset_handler<C: Context>(
+    _session: AdminSession,
+    ctx: web::Data<C>,
+    create: web::Json<CreateSystemDataset>,
+) -> Result<web::Json<IdResponse<DatasetId>>>
+where
+    C::Session: MockableSession,
+{
+    let create = create.into_inner();
+    let mut definition = create.definition;
+
+    let volumes = get_config_element::<Data>()?.volumes;
+    let volume_path = volumes
+        .get(&create.volume)
+        .ok_or(error::Error::UnknownVolumeId)?;
+    let volume = Volume {
+        id: create.volume,
+        path: volume_path.clone(),
+    };
+
+    adjust_meta_data_path(&mut definition.meta_data, &volume)?;
```

Comment:
Can we prohibit both?

Reply:
we do, with `canonicalize_subpath` and `path_with_base_path`

Reply:
But we write it down as a full path in the dataset. Can't we have it relative to a volume?

Reply:
it's not a bad idea to do this. This would also allow us to move whole volumes to a different path. We would have to change all of the MetaData definitions though and make the `path` two-parts: 1. volume/upload ref 2. path relative to volume/upload. And then when computing the resultof  `loading_info` turn it into an absolute path that gdal then uses. We also would have to handle the case of external resources differently I guess, because how do URLs fit into this? and the virtual filesystems of gdal that come before the file path? Then we would also need to adjust all of the existing definition files.

I would propose to do this in a separate task.

Reply:
I guess you are right.